lib/profile.dart

```diff
@@ -0,0 +1,150 @@
+import 'package:flutter/material.dart';
+import 'pages/data.dart';
+
+class ProfilePage extends StatefulWidget {
+  const ProfilePage({super.key, required this.title});
+
+  final String title;
+
+  @override
+  State<ProfilePage> createState() => _ProfilePageState();
+}
+
+
+class _ProfilePageState extends State<ProfilePage> {
+
+  String User_name_data = '読み込み中...';
+  int grade = 0;
+  String User_info = '読み込み中...';
+  String User_link = '読み込み中...';
+
+  @override
+  void initState() {
+    super.initState();
+    ProfileDataFetcher();
+    fetchProfileData();
+  }
+
+  //データをフェッチして変数に格納する関数
+  void fetchProfileData() async {
+    ProfileDataFetcher fetcher = ProfileDataFetcher();
+    try {
+      final data = await fetcher.fetchData();
+      debugPrint('Fetched data: $data');
+      setState(() {
+        User_name_data = data['name'];
+        grade = data['grade'];
+        User_info = data['bio'];
+        User_link = data['sns']['x'];
+      });
+    } catch (e) {
+      setState(() {
+        User_name_data = "エラー";
+        grade = 0;
```

Comment:
明示的にエラーな値があるといいよね

```suggestion
        grade = -1;
```

将来的にはEnumの方が好ましい

Reply:
とりあえず-1を代入しておきました